File: app/src/main/java/com/privat/contacts/domain/model/SubscriptionDomain.java
package com.privat.contacts.domain.model;

import org.jetbrains.annotations.NotNull;

public interface SubscriptionDomain {
    @NotNull
    <T> T map(Mapper<T> mapper);

    interface Mapper<T> {
        @NotNull
        T map(
                int userId,
                String plan,
                String status,
                String paymentMethod,
                String term);
    }
}

File: app/src/main/java/com/privat/contacts/domain/model/SubscriptionDomain.kt
package com.privat.contacts.domain.model

interface SubscriptionDomain {
    fun <T> map(mapper: Mapper<T>): T
    interface Mapper<T> {
        fun map(
                userId: Int,
                plan: String,
                status: String,
                paymentMethod: String,
                term: String): T
    }
}
